File: docs/plugin-starter/ts/test/main.test.ts
import test from "node:test";
import assert from "node:assert/strict";

import { before_step } from "../src/main";

test("returns allow for non-targeted step", async () => {
  const result = await before_step({
    tenant_id: "tnt_001",
    workflow: { job_id: "job_1", step: "suspend_service" },
    config: { target_step: "activate_service", auto_allow: true }
  });

  assert.equal(result.decision, "allow");
  assert.equal(result.reason, "step_not_targeted");
});

test("requires manual approval when auto_allow disabled", async () => {
  const result = await before_step({
    tenant_id: "tnt_001",
    workflow: { job_id: "job_2", step: "activate_service" },
    config: { target_step: "activate_service", auto_allow: false }
  });

  assert.equal(result.decision, "require_manual_approval");
  assert.equal(result.reason, "manual_review_required");
});

File: docs/plugin-starter/elixir/test/main_test.exs
defmodule MainTest do
  use ExUnit.Case, async: true

  test "denies when hour outside allowlist" do
    result =
      Main.pre_authorize(%{
        "request" => %{"timestamp_utc" => "2026-01-01T03:00:00Z"},
        "config" => %{"allowed_hours_utc" => [9, 10, 11]}
      })

    assert result["decision"] == "deny"
    assert result["reason"] == "outside_allowed_hours"
  end
end
